app/controllers/export/comments_controller.rb

```diff
@@ -0,0 +1,10 @@
+module Export
+  class CommentsController < ApplicationController
+    def index
+        comments = Comment.joins(:post).where(status: "public", comments: {status: "public"})
```

Comment:
`status: "public"` also checks the comment's status:
```
 _ = Comment.joins(:post).where(status: "public", comments: {status: "public"})
  Comment Load (0.4ms)  SELECT "comments".* FROM "comments" INNER JOIN "posts" ON "posts"."id" = "comments"."post_id" WHERE "comments"."status" = ? AND "comments"."status" = ?  [["status", "public"], ["status", "public"]]
```

Perhaps you meant to write this?
```suggestion
        comments = Comment.joins(:post).where(status: "public", posts: {status: "public"})
```

This should explain why I'm getting comments of archived posts in the export.

Reply:
Let's talk about what "visible" means. Private comments on private posts are also visible on the website.

Reply:
Add a select/pluck to make sure that we're only loading data we actually need, to reduce database-application traffic.

```suggestion
        comments = Comment.joins(:post).where(status: "public", posts: {status: "public"}).pluck(:'posts.title', :'comments.author', :'comments.body')
```